src/bin/commands/containers.rs

```diff
@@ -0,0 +1,502 @@
+        list_containers(self, &engine)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct RemoveAllContainers {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Force removal of containers.
+    #[clap(short, long)]
+    pub force: bool,
+    /// Remove containers. Default is a dry run.
+    #[clap(short, long)]
+    pub execute: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl RemoveAllContainers {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        remove_all_containers(self, &engine)
+    }
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Containers {
+    /// List cross containers in local storage.
+    List(ListContainers),
+    /// Stop and remove cross containers in local storage.
+    RemoveAll(RemoveAllContainers),
+}
+
+impl Containers {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        match self {
+            Containers::List(args) => args.run(engine),
+            Containers::RemoveAll(args) => args.run(engine),
+        }
+    }
+
+    pub fn engine(&self) -> Option<&str> {
+        match self {
+            Containers::List(l) => l.engine.as_deref(),
+            Containers::RemoveAll(l) => l.engine.as_deref(),
+        }
+    }
+
+    pub fn verbose(&self) -> bool {
+        match self {
+            Containers::List(l) => l.verbose,
+            Containers::RemoveAll(l) => l.verbose,
+        }
+    }
+}
+
+fn get_cross_volumes(engine: &docker::Engine, verbose: bool) -> cross::Result<Vec<String>> {
+    let stdout = docker::subcommand(engine, "volume")
+        .arg("list")
+        .arg("--format")
+        .arg("{{.Name}}")
+        .arg("--filter")
+        // handles simple regex: ^ for start of line.
+        .arg("name=^cross-")
+        .run_and_get_stdout(verbose)?;
+
+    let mut volumes: Vec<String> = stdout.lines().map(|s| s.to_string()).collect();
+    volumes.sort();
+
+    Ok(volumes)
+}
+
+pub fn list_volumes(
+    ListVolumes { verbose, .. }: ListVolumes,
+    engine: &docker::Engine,
+) -> cross::Result<()> {
+    get_cross_volumes(engine, verbose)?
+        .iter()
+        .for_each(|line| println!("{}", line));
+
+    Ok(())
+}
+
+pub fn remove_all_volumes(
+    RemoveAllVolumes {
+        verbose,
+        force,
+        execute,
+        ..
+    }: RemoveAllVolumes,
+    engine: &docker::Engine,
+) -> cross::Result<()> {
+    let volumes = get_cross_volumes(engine, verbose)?;
+
+    let mut command = docker::subcommand(engine, "volume");
+    command.arg("rm");
+    if force {
+        command.arg("--force");
+    }
+    command.args(&volumes);
+    if volumes.is_empty() {
+        Ok(())
+    } else if execute {
+        command.run(verbose, false).map_err(Into::into)
+    } else {
+        eprintln!("note: this is a dry run. to remove the volumes, pass the `--execute` flag.");
```

Comment:
Probably a good idea for consistency, since I think we use it for warnings. The only exceptions are in this PR, so definitely. We use capitalization for `eprintln!("Warning: ...");`. so this should be changed.